google-cloud-storage/src/main/java/com/google/cloud/storage/BlobInfo.java

```diff
@@ -1095,8 +1095,10 @@ public ObjectAccessControl apply(Acl acl) {
     if (retentionExpirationTime != null) {
       storageObject.setRetentionExpirationTime(new DateTime(retentionExpirationTime));
     }
-
-    storageObject.setKmsKeyName(kmsKeyName);
+    storageObject.setKmsKeyName(
+        kmsKeyName != null && kmsKeyName.contains("/cryptoKeyVersions/")
+            ? kmsKeyName.substring(0, kmsKeyName.indexOf("/cryptoKeyVersions/"))
+            : kmsKeyName);
     storageObject.setEventBasedHold(eventBasedHold);
```

Comment:
a nit: `"/cryptoKeyVersions/"` is used twice, you need either introduce a constant to keep this value, or use the code like:
```
int versionIndex = kmsKeyName == null ? -1 : kmsKeyName.indexOf("/cryptoKeyVersions/")
if (versionIndex >= 0) {
   kmsKeyName = kmsKeyName.substring(0, versionIndex);
}
```

Reply:
done